packages/monaco/src/typings/monaco/index.d.ts

```diff
@@ -1012,28 +1039,28 @@ declare module monaco.referenceSearch {
 
 declare module monaco.quickOpen {
 
-    // https://github.com/theia-ide/vscode/blob/standalone/0.20.x/src/vs/base/browser/ui/inputbox/inputBox.ts#L59
+    // https://github.com/theia-ide/vscode/blob/standalone/0.23.x/src/vs/base/browser/ui/inputbox/inputBox.ts#L58
     export interface IMessage {
         content: string;
         formatContent?: boolean; // defaults to false
         type?: 1 /* INFO */ | 2  /* WARNING */ | 3 /* ERROR */;
     }
 
-    // https://github.com/theia-ide/vscode/blob/standalone/0.20.x/src/vs/base/browser/ui/inputbox/inputBox.ts#L91
+    // https://github.com/theia-ide/vscode/blob/standalone/0.23.x/src/vs/base/browser/ui/inputbox/inputBox.ts#L90
     export class InputBox {
         inputElement: HTMLInputElement;
         setPlaceHolder(placeHolder: string): void;
         showMessage(message: IMessage, force?: boolean): void;
         hideMessage(): void;
     }
 
-    // https://github.com/theia-ide/vscode/blob/standalone/0.20.x/src/vs/base/parts/tree/browser/tree.ts#L17
+    // https://github.com/theia-ide/vscode/blob/standalone/0.20.x/src/vs/base/parts/tree/browser/tree.ts#L17 - not found in 0.23.x
     export interface QuickOpenTree {
         onDidChangeFocus: monaco.Event<any>;
         getFocus(): monaco.quickOpen.QuickOpenEntry;
     }
```

Comment:
Should we remove those missing typings? This file is supposed to describe the `monaco` library namespace, so leaving those might become confusing?

Reply:
@marechal-p That is a good point - my thought here was to update references and mandatory changes, and not change interface with removed methods - as I don't know the implications of that - and also didn't add missing methods as obviously currently they aren't used - but I do not know what is preferred way of going about it - @vince-fugnitto ?

Reply:
What is the expected next step after this change? It would help to see how this change relates to the finished work as from what I understand the current PR updates typings for `monaco.*` packages. If we lie in our typings it will lead to us trying to use undefined symbols when using those APIs internally. Tell me if I misunderstood anything.

Reply:
@akosyakov you might be busy but do you mind confirming this? Those typings are meant to describe what we can use from the updated `monaco` APIs right? So we don't want to define erroneous/missing typings?

Reply:
If some type, for example, `QuickOpenTree`, is not found for `0.23` version of `Monaco` - we can not use it on the `Theia` side. 
I believe we should:
- remove the type from the `index.d.ts` file
- detect places where the missing type is used on the `Theia` side
- investigate what was changed for the new version of `Monaco`
- adapt detected places in `Theia` to the new version of the `Monaco`

Reply:
Another very important kind of work here:
Check semantic for all types defined in `index.d.ts` file.

The point is:
-  `index.d.ts` file contains types which we use for `Theia`
- it's OK to define, for example, only one method for a type in `index.d.ts` if on the `Theia` side we use only this method of the type and don't use other methods and properties of the type
- but if some type is defined in `index.d.ts` file - at the current step it's important to check that a type still present in the new version of the `Monaco`, the semantic is still correct, for example, it still has a method `someMethodName(a: string, b: number)` and still expect 2 parameters with some expected types.

Reply:
@RomanNikitenko 
@amiramw 

> If some type, for example, `QuickOpenTree`, is not found for `0.23` version of `Monaco` - we can not use it on the `Theia` side.
> 
> I believe we should:
> 
> * remove the type from the `index.d.ts` file
> * detect places where the missing type is used on the `Theia` side
> * investigate what was changed for the new version of `Monaco`
> * adapt detected places in `Theia` to the new version of the `Monaco`

After digging some more, it seems that all missing types in Theia are related to the `quickOpen` functionality. Checking VS Code I found a few commits related to `quickopen` like: https://github.com/microsoft/vscode/commit/572d9c6a87eeada2afc6bf695278fe9074283f8f
https://github.com/microsoft/vscode/commit/f6fba38e9a43df3a1b90f26cad89da19562db13c
(and more)

IMO making this change you are requesting here is not a trivial one and we should consider opening a different PR for upgrading `quickopen` module to latest implementation - as it would need investigation and diving in to understand changes done in VS Code and also decide which ones are relevant to Theia. 

What do you think?

Reply:
I understand that it's additional efforts to fix the problem.
But how we can upgrade the Monaco version without it?

Reply:
@RomanNikitenko You are right...

Reply:
> I understand that it's additional efforts to fix the problem.
But how we can upgrade the Monaco version without it?

@marcdumais-work @vince-fugnitto @RomanNikitenko @marechal-p  @offer8

Hi,
In regard to "replacing" `quickOpen` with `quickInput`, I'm making progress but still have a way to go and so I want to inform you that this is going to be a substantial change which I'm somewhat hesitant with as it would require testing/checking/reviewing of all existing functionality currently implemented with `quickOpen`. 

I have to mention that there could be an alternative option which would be something like "copying" `quickOpen` code from 0.20.x to Theia so as not to use `monaco.quickOpen`. That is in order for current `quickOpen` functionality to continue working as it is without change (I haven't started this route yet).

It could take me a week or two to complete the `quickInput` implementation and I would be obviously better equipped to point out how confident I am with the new implementation. Currently I keep adding more and more 0.23.x APIs as `quickInput, quickAccess, listWidget, pickerQuickAccess` and removing existing `quickOpen` code like `PrefixQuickOpenService, MonacoQuickOpenService` etc.

I would like to hear your point of view in the matter.

Thanks,
Dan

Reply:
As far as I remember from the previous upgrade - `quickOpen` functionality - it's an interaction `Theia` side and `Monaco` side. 
Based on this understanding and fact that `quickOpen` related types were removed on the `Monaco` side, I don't see how we can avoid alignment with `Monaco` objects.
On the other hand, I understand, that the mentioned alignment to `quickInput` objects requires significant changes on the `Theia` side. 
So, could you provide more details about your idea:
> I have to mention that there could be an alternative option which would be something like "copying" quickOpen code from 0.20.x to Theia so as not to use monaco.quickOpen.